File: recovered/ceil.py
import numpy as np, eldata as ed, itertools
from itertools import combinations
from collections import Counter
CAP=lambda l: 2*(2*l+1)
occ=set(ed.E.values())
def leq(a,b): return all(x<=y for x,y in zip(a,b))

def build(NMAX,LMAX):
    return [(n,l,k) for n in range(1,NMAX+1) for l in range(0,min(n,LMAX+1))
            for k in range(1,CAP(l)+1)]
def props(NMAX,LMAX):
    L=build(NMAX,LMAX); S=set(L)
    jn=lambda a,b: tuple(max(x,y) for x,y in zip(a,b))
    mt=lambda a,b: tuple(min(x,y) for x,y in zip(a,b))
    # closure
    cl=all(jn(a,b) in S and mt(a,b) in S for a,b in combinations(L,2))
    # 2n^2
    twon2=all(sum(CAP(l) for l in range(0,min(n,LMAX+1)))==2*n*n for n in range(1,NMAX+1))
    # parity
    ev=sum(1 for c in L if sum(c)%2==0)
    par=(ev==len(L)-ev)
    # rank symmetry
    rk=Counter(sum(c) for c in L)
    seq=[rk[i] for i in range(min(rk),max(rk)+1)]
    sym=(seq==seq[::-1])
    # min element
    mn=min(L)
    # reserved cells (occupied set fixed)
    res=sum(1 for c in L if c not in occ)
    # columns
    cols=[(n,l) for n in range(1,NMAX+1) for l in range(0,min(n,LMAX+1))]
    occols=set((n,l) for (n,l,k) in occ)
    # down-sets of column poset
    def cleq(a,b): return a[0]<=b[0] and a[1]<=b[1]
    ds=[0]
    order=sorted(cols)
    def enum(i,ch):
        if i==len(order): ds[0]+=1; return
        c=order[i]; enum(i+1,ch)
        if all(d in ch for d in cols if cleq(d,c) and d!=c): enum(i+1,ch|{c})
    if len(cols)<=30: enum(0,frozenset())
    else: ds[0]=None
    # ideal property of occupied set within this lattice
    ideal=not any(leq(y,x) and y not in occ for x in occ for y in L)
    # maximal occupied + covering
    maxel=[x for x in occ if not any(leq(x,y) and x!=y for y in occ)]
    nxt=set()
    for m in maxel:
        for d in range(3):
            c=list(m); c[d]+=1; c=tuple(c)
            if c in S and c not in occ: nxt.add(c)
    # Sperner: largest rank level
    return dict(cells=len(L), closed=cl, twon2=twon2, parity=par, ranksym=sym,
                minel=mn, reserved=res, cols=len(cols), downsets=ds[0],
                ideal=ideal, maxel=len(maxel), covering=len(nxt),
                maxrank=max(rk.values()))

print("="*100)
print("CEILING SENSITIVITY: which results survive a change of n_max and l_max?")
print("="*100)
cfgs=[(7,4),(8,4),(7,5),(8,5),(9,4),(7,3)]
res={}
for c in cfgs: res[c]=props(*c)
keys=['cells','closed','twon2','parity','ranksym','minel','reserved','cols',
      'downsets','ideal','maxel','covering','maxrank']
print(f"  {'property':<22}"+"".join(f"{str(c):>12}" for c in cfgs))
print("  "+"-"*(22+12*len(cfgs)))
for k in keys:
    vals=[res[c][k] for c in cfgs]
    inv = len(set(map(str,vals)))==1
    tag='  INVARIANT' if inv else '  contingent'
    print(f"  {k:<22}"+"".join(f"{str(v):>12}" for v in vals)+tag)

File: recovered/ceil.js
B.push(h3('10.5  Which results depend on the ceilings'));
B.push(p([t('The bounds n ≤ 7 and ℓ ≤ 4 in Equation (1) are modelling choices reflecting the elements presently known, not features of the physics. It is therefore necessary to say which of the results above survive a change in those bounds and which do not, since a property of a truncation is a weaker thing than a property of the construction. Every result in this part was recomputed under six ceiling configurations, from (n ≤ 7, ℓ ≤ 3) to (n ≤ 9, ℓ ≤ 4) and (n ≤ 8, ℓ ≤ 5), with the occupied set held fixed. The outcome divides cleanly.')]));
B.push(tcap(3,'Sensitivity of each result to the shell and subshell ceilings, recomputed across six configurations with the occupied set held fixed. Invariant results hold in every configuration tested; contingent results take a value specific to the chosen bounds.'));
B.push(table(
  ['Result','Section','Status','Value at (7, 4)'],
  [['Λ is closed under join and meet','8','invariant','—'],
   ['Unique minimum is (1, 0, 1)','8','invariant','hydrogen'],
   ['Rank sequence is not palindromic','10.4','invariant','—'],
   ['Parity balance F(−1) = 0','10.3','invariant','105 / 105'],
   ['Occupied set is an order ideal','13','invariant','—'],
   ['Occupied set has three maximal cells','16','invariant','Lr, Cn, Og'],
   ['Admissible cell count','3','contingent','210'],
   ['Reserved cell count','3','contingent','92'],
   ['Column count','14','contingent','25'],
   ['Down-sets of the column poset','14','contingent','120'],
   ['Covering cells at the frontier','16','contingent','5'],
   ['Largest rank level (Sperner value)','10.2','contingent','15'],
   ['Σ 2(2ℓ+1) = 2n²','10.1','holds where ℓ ≤ n−1 does not bind','n ≤ 5']],
  [2900,900,1900,1500]));
B.push(p([t('The invariant results are the ones this paper rests on most heavily. Closure, the identity of the minimum element, the parity balance, the rank asymmetry and the order-ideal property all hold under every configuration tested, and in three cases the reason is a proof rather than a computation: closure follows from the form of the constraints, the minimum follows from the constant floors, and the parity balance follows from every column capacity being even.')],
  {spacing:{before:160,after:120}}));
B.push(p([t('The contingent results are counts, and each should be read as a count for the lattice as bounded here. That the reserved cells number 92, that the column poset has 120 down-sets, that the frontier offers five covering cells and that the largest antichain has 15 elements are all statements about (n ≤ 7, ℓ ≤ 4). Extending the lattice to n ≤ 8 raises them to 142, 219, six and 18 respectively. The qualitative claims built on them survive — reserved cells still concentrate at high n and high ℓ, the realised system is still one down-set among many, the frontier is still a small finite set — but the numbers do not transfer, and the paper states them as figures for this truncation rather than as invariants.')]));
B.push(p([t('One entry requires a correction to what was said earlier. Section 10.4 attributed the failure of rank symmetry to the interaction of the two ceilings. That explanation is wrong. The rank sequence remains non-palindromic when the ℓ ceiling is lifted entirely, tested up to n ≤ 20 with ℓ ≤ n−1 and 5,740 cells, so the asymmetry is a property of the constraint structure itself — the growth of the k-range with ℓ against the bounding of ℓ by n — and not of the truncation. The conclusion of Section 10.4 stands; its stated reason did not, and has been amended there.')]));
